core/io/resource.cpp

```diff
@@ -124,6 +124,9 @@ String Resource::generate_scene_unique_id() {
 	return id;
 }
 
+// Warning: there is no validation here.
+// Use is_ascii_identifier_char() to validate before passing it in.
+// Example: if you pass in a quote, it will corrupt the scene when saving.
 void Resource::set_scene_unique_id(const String &p_id) {
 	scene_unique_id = p_id;
```

Comment:
Is there an escape we can do that will let us use arbitrary strings but avoid breaking the scene?

Reply:
I do reiterate my original concern in https://github.com/godotengine/godot/pull/88111#discussion_r1485169534. That is, the method exposed to outside developers should not be able to brick scenes this way (see comment for more).